python/tvm/relay/transform/quantize/_calibration_callback.py

```diff
@@ -0,0 +1,106 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import numpy as np
+
+
+class CalibrationCallback:
+    """Abstract class that defines the API calibration methods."""
+
+    def calibrate_pattern(self, calibration_info):
+        """Calculates the scale and zero points for quantizing parts
+        of a generic pattern. If you would like to do per-channel or
+        pattern-specific calibration, please overwrite calibrate_pattern
+        in the relevant QuantizerPattern.
+
+        Parameters
+        ----------
+        calibration_info : CalibrationInfo
+            The class containing relevant information and utility functions
+            to calibrate one instance of a pattern.
+
+        Returns
+        -------
+        scale_zp_map : Dictionary
+            A map from the names of scales and zero point variables in this
+            pattern to their values.
+        """
+        raise NotImplementedError
+
+
+class GlobalCalibrationCallback(CalibrationCallback):
+    """Sets the scales and zero points to a user-provided value."""
+
+    def __init__(self, scale_value, zp_value):
+        self.scale_value = np.array(scale_value).astype("float32")
+        self.zp_value = np.array(zp_value).astype("int32")
+
+    def calibrate_pattern(self, calibration_info):
+        """Returns the scale and zero point value set during initialization to
+        the Calibrater."""
+        scale_zp_values = {}
+        for i in range(len(calibration_info.input_scale_zps)):
+            scale_name = calibration_info.input_scale_zps[i][0].name_hint
+            scale_zp_values[scale_name] = self.scale_value
+            zp_name = calibration_info.input_scale_zps[i][1].name_hint
+            scale_zp_values[zp_name] = self.zp_value
+
+        return scale_zp_values
+
+
+class AverageMaxCalibrationCallback(CalibrationCallback):
+    """Calculates scales and zero points by calculating the average of the maxiumum
+    absolute value of the data we are quantizing."""
```

Comment:
You're calling this average Max, but you're doing min and max ?

Reply:
I was thinking because it's the average of the max and abs(min), it kind of worked. AverageMinMax seemed a bit long.. I'm open to other naming suggestions, or we could go with AverageMinMax